src/main/java/impl/image/ImageConverterImpl.java

```diff
@@ -0,0 +1,171 @@
+                colorValueGetter = this::deconstructColor;
+            }
+        }
+        return colorValueGetter;
+    }
+
+    /**
+     * Works sorta like a one-time generated property with backing field
+     * Gets a function that sets value field of a Color object.
+     * If reflection access was OK - it's direct setting to the private field of a freshly created Color object
+     * The fallback just does what's obvious approach - gets different bit parts and constructs proper object
+     */
+    private Function<Integer, Color> getColorValueSetterFunction() {
+        if (colorValueSetter == null) {
+
+            try {
+                var field = Color.class.getDeclaredField("value");
+
+                if (valueIsPrivate) {
+//                  Yeah, the fun part about java is that if you get the field, `setAccessible` will never throw
+                    field.setAccessible(true);
+                    valueIsPrivate = false;
+                }
+
+                colorValueSetter = (Integer value) -> {
+                    try {
+                        var color = new Color(0);
+                        field.set(color, value);
+                        return color;
+                    } catch (IllegalAccessException e) {
+                        e.printStackTrace();
+//            Weird fallback, for some reason there's no way here to use it as a functor and apply like w'd really want.
+                        return constructColor(value);
+                    }
+                };
+            } catch (NoSuchFieldException e) {
+                colorValueSetter = this::constructColor;
+            }
+        }
+        return colorValueSetter;
+    }
+
+
+    /**
+     * Transforms 2d array of type S to 2d array of type T
+     * @param <S> Source type
+     * @param <T> Target type
+     */
+    private static class ArrayTransformer<S, T> {
+
+        /**
+         * Transforms 2d array of type S to 2d array of type T using given converter function that is applied element-wise
+         * @param source source 2d array of type S
+         * @param converter converter function that is applied to each element
+         * @param targetClass Class object of target type, needed to be able to access target ctor cause of type erasure.
+         * @return converted 2d array of type T
+         * @implNote probably not the best looking impl since I can imagine same s possible using streams but
+         * I couldn't get that right...
+         */
+        @SuppressWarnings("unchecked")
+        public T[][] transform(S[][] source, Function<S, T> converter, Class<T> targetClass) {
+//            T[][] result = new T[source.length][]; //oh, yeah, type erasure, I can't do this...
+//            Ok, there's sorta workaround we can pass target class obj...
+//            And since we need nested Arrays... and Class<T[]> is not a thing.....
+            Class targetArrayClass = Array.newInstance(targetClass, 0).getClass();
```

Comment:
`Class` is a parametrized one, so don't use raw types in such cases

Reply:
There's a comment right above it saying that I couldn't' parametrize Class object properly for it to store array class, If you could help me do it more properly - I'll be glad to do this. _like if there is something proper in my approach to this task..._

Reply:
If you can't (don't want to) provide any types for parametrized one, you should use wildcard `?`, so use `Class<?>` here